lib/system_message.rb

```diff
@@ -43,6 +43,8 @@ def create(type, params = {})
 
     post = I18n.with_locale(@recipient.effective_locale) { creator.create }
 
+    DiscourseEvent.trigger(:system_message_sent, post_id: post.id, topic_id: post.topic.id, message_type: type)
```

Comment:
Let's just make the entire `post` object available to the trigger. I've found that to be far more versatile because we can support all form of use cases. 

 ```suggestion
    DiscourseEvent.trigger(:system_message_sent, post: post, message_type: type)
```